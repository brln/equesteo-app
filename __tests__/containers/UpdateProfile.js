import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import UpdateProfileContainer from '../../containers/UpdateProfile'


const mockStore = configureStore([thunk])

test('render', () => {
  const store = mockStore(fromJS({
    pouchRecords: {
      horseUsers: {},
      users: {
        userID1: {
          _id: 'userID1'
        }
      },
      userPhotos: {}
    },
    localState: {
      userID: 'userID1'
    },
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <UpdateProfileContainer />
    </Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})

