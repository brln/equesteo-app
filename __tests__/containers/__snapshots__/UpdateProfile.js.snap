// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="always"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "padding": 5,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              header={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": null,
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 15,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                }
              }
            >
              <Text>
                First Name:
              </Text>
            </View>
            <View
              cardBody={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderRadius": 2,
                    "flexDirection": "row",
                    "paddingBottom": -5,
                    "paddingLeft": -5,
                    "paddingRight": -5,
                    "paddingTop": -5,
                  },
                  Object {
                    "marginLeft": 20,
                    "marginRight": 20,
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                maxLength={200}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#4b9ec1",
                    "borderWidth": 1,
                    "height": 50,
                    "padding": 10,
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              header={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": null,
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 15,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                }
              }
            >
              <Text>
                Last Name:
              </Text>
            </View>
            <View
              cardBody={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderRadius": 2,
                    "flexDirection": "row",
                    "paddingBottom": -5,
                    "paddingLeft": -5,
                    "paddingRight": -5,
                    "paddingTop": -5,
                  },
                  Object {
                    "marginLeft": 20,
                    "marginRight": 20,
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                maxLength={200}
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#4b9ec1",
                    "borderWidth": 1,
                    "height": 50,
                    "padding": 10,
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              header={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": null,
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 15,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                }
              }
            >
              <Text>
                 About Me: 
              </Text>
            </View>
            <View
              cardBody={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderRadius": 2,
                    "flexDirection": "row",
                    "paddingBottom": -5,
                    "paddingLeft": -5,
                    "paddingRight": -5,
                    "paddingTop": -5,
                  },
                  Object {
                    "marginBottom": 20,
                    "marginLeft": 20,
                    "marginRight": 20,
                  },
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                maxLength={2000}
                multiline={true}
                onChangeText={[Function]}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#4b9ec1",
                    "borderWidth": 1,
                    "height": 150,
                    "padding": 10,
                    "textAlignVertical": "top",
                    "width": "100%",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
